core/dbt/context/base.py

```diff
@@ -248,9 +249,14 @@ def add_macros_from(
             # adapter packages are part of the global project space
             _add_macro_map(context, package_name, macro_map)
 
-            if package_name == self.search_package_name:
+            if package_name == config.project_name:
```

Comment:
Instead of changing the method signature, please just use `self.config.project_name` here!

Reply:
Ah, I didn't realise we have a `config` object in the class already. Great idea.